dataplicity/m2m/bencode.py

```diff
@@ -60,12 +61,19 @@ def encode(obj):
 
     def add_encode(obj):
         if isinstance(obj, bytes):
-            append(b"{}:{}".format(len(obj), obj))
+            if PY2:
+                append(b"{}:{}".format(len(obj), obj))
+            elif PY3:
+                append(six.b("{}:".format(len(obj))))
+                append(obj)
         elif isinstance(obj, text_type):
             obj = obj.encode('utf-8')
-            append(b"{}:{}".format(len(obj), obj))
+            if PY2:
+                append(b"{}:{}".format(len(obj), obj))
+            elif PY3:
+                add_encode(obj)
```

Comment:
the `obj` will be `bytes` when using python3 (notice the above if PY2 / PY3). It will be text_type in PY2, but this is handled in line 72. I could add more verbose code here, but since it would be a carbon copy of what's in lines 64-68, I decided to take this route.